### ISSUE TITLE
navigation.js:173 Uncaught TypeError

### ISSUE THREAD

**Issue by garrison0** [gatsbyjs/gatsby#21263]
Still getting this error from links after updating to Gatsby 2.19.12. I've checked my dependencies and re-installed node modules.

I can recreate this issue by:

1. updating [this](https://github.com/thomaswang/gatsby-personal-starter-blog) starter
2. making a new root element implementing the router with '/' corresponding to 'index.js', '/foo' corresponding to the blog or some trivial page
3. putting a new link to the blog or some trivial page onto index.js and clicking it

```
navigation.js:173 Uncaught TypeError: Cannot read property 'getElementsByTagName' of null
    at navigation.js:173
```

```
>  170 | if (document.title) {
>  171 |      pageName = document.title
>  172 | }
>  173 | const pageHeadings = document
>  174 |     .getElementById(`gatsby-focus-wrapper`)
>  175 |     .getElementsByTagName(`h1`)
>  176 | if (pageHeadings && pageHeadings.length) {
```

See [this previously closed issue](https://github.com/gatsbyjs/gatsby/issues/21139), seems other people are still running into this aswell

**Comment by marcysutton** [comment]
Hi there, I got this to work with the latest Gatsby version by deleting the `node_modules` folder and package-lock.json (or yarn.lock) and then doing a reinstall.

**Comment by garrison0** [comment]
The problem persists for me in this environment.

I've deleted both node_modules and package-lock.json, reinstalled, and checked the dependencies. I've noticed gatsby-link still requires @reach/router 1.2.6 whereas gatsby requires 1.2.1. I'm not sure how relevant that is because the bug persists whether I important Link from gatsby, gatsby-link, or @reach/router. 

Interesting if I build and serve, when I first try to load the link, it works, but afterwards I get the error in the devtools console.

If there's any other information I could provide, I'd be happy to

**Comment by arun-is** [comment]
I have the same exact issue, with the same steps that @garrison0 listed out.

**Comment by arun-is** [comment]
@garrison0 what is the file name that you are using for the new page? Are you sure it matches that path that you are linking to? For example, did you make a `foo.js` and then link to `/foo`? I found if I make a `Foo.js` and then link to `/foo`, I see the error you are seeing

**Comment by garrison0** [comment]
@arun-is 

that fixed the problem, thanks

but why wouldn't something like this work? : 

```
import NewPage from './NewPage.js'

const App = () => (
    <Router>
      <Home path="/" />
      <NewPage path="/newpage" /> 
    </Router>
)

export default App;
```

**Comment by sidharthachatterjee** [comment]
Looks like the issue here is that page paths are case sensitive and hence it isn't able to find your page and breaks at this confusing spot (because it tries to announce the route change). 

We should catch that before attempting this and redirect to 404.

**Comment by jakubzloczewski** [comment]
@garrison0, have you found solution for client-only routes? I have the same issue with latest version.

**Comment by nickwarters** [comment]
I've experienced the same issue, but only with some links. After trying a number of things I found it was only an issue with url paths with a capital letter in them. As soon as I switch them all lowercase they work fine. 
To get around this, I have tried using a slugify function to convert all each everything to a lowercase URL in my Gatsby-Node but it doesn't seem to be wanting to play ball.

**Comment by nickwarters** [comment]
Okay ... I managed to fix it for me. I had pages which were created with slugified paths, however some of my links were pointing to un-slugified paths of the pages. Theoretically this should have just returned a 404 Page not Found however instead it threw this error. 

Resolved for me now though.

**Comment by cmr624** [comment]
Having a similar issue, I am using Gatsby-Image, and reloading or hot reloads crash with the error.

How would I go about reverting to the correct version of @reach/router ? I have Gatsby installed, but no @reach/router in my `package.json`

**Comment by jlabs** [comment]
Having the same issue using the same starter template.

For me, removing the line `<MDXRenderer>{post.body}</MDXRenderer>` in src/templates/blog-post.js made all the pages under /blog load instantly. I'd not made any changes to the starter project up to this point (other than fixing all the errors OOTB). The content on each blog post page still loads, and there's no console errors. Yet.

**Comment by Ir1d** [comment]
I'm having the same issue on certain page. There doesn't seem to be much difference between the pages. Also using image sharp

**Comment by xanderios** [comment]
Having the same issue.
OS: Windows 10,
*"gatsby": "^2.19.12"*.

**Expected behaviour:**
Link between pages with Gatsby Link as usual.

**Actual behaviour:**
Window loads with div *#___gatsby*, but without div *#gatsby-focus-wrapper*, as seen in *navigation.js*:
```javascript
const pageHeadings = document
  .getElementById(`gatsby-focus-wrapper`)
  .getElementsByTagName(`h1`)
if (pageHeadings && pageHeadings.length) { ...
```

I have tried installing *@reach/router 1.2.1* , checked all *src/pages* filenames, all of them slugs, with correct *Link* paths to them.

I'll add any additional information if needed.

**Comment by xanderios** [comment]
Had the same issue, not a Gatsby issue I guess:
What I did to debug that:
1. Commented every section of my homepage *(the page that gave error on build)*;
2. Uncomment sections of homepage 1 by 1 until I find the one with error.

In my case it was a *SwiperJS* Instance that was removed between pages, and that was causing the error.

I read that some people installed some plugins and libraries and got the error, so I figured that my *Swiper* (but other libraries too) could bring the error. Hope it helps you guys.

**Comment by saadaouad** [comment]
Any update on this please? still getting the same issue using `"gatsby": "2.20.9",` version

**Comment by EliranGooner** [comment]
Still getting this error too.

**Comment by dannyvaughton** [comment]
I am also getting the same error on gatsby@2.20.9

It appears to be when a <a> is being used by one of my antd components. I can't remove these easily as they have been integral to my application.

Also, the error seemed to pop out of nowhere. One second it was working, the next not.

I have cleared the cache, removed node_modules & lock file. Not sure where to go from here.

**Comment by shicholas** [comment]
I've been getting this error too trying to create dynamic client-side routes using Reach Router (fixed to version 1.2.1 to address other comments in this thread.

I pushed https://github.com/gatsbyjs/gatsby/pull/23196 as a potential fix to this nullcheck.

**Comment by BrooksPoltl** [comment]
I had this issue, and nothing else worked until I found out that slugs are formatted `/article-name/` so I was creating a page for `/topic//article-name/` by doing `createPage({
      path: `${post.node.frontmatter.topic}/${post.node.fields.slug}`,
      component: blogPost,
      context: {
        slug: post.node.fields.slug,
        previous,
        next,
      },
    })` Which when you link to this page it will resolve to `topic/article-name`

**Comment by Hyraze** [comment]
This issue will be closed with this pull request #23196
